modules/Core.js

```diff
@@ -56,31 +56,38 @@ const matchPath = (pathname, path, exact = false) => {
 /**
  * A higher-order component that starts listening for location
  * changes (calls `history.listen`) and re-renders the component
- * each time it does. Also, passes `context.history` as a prop.
+ * each time it does. Also, passes `context.router` as a prop.
  */
-const withHistory = (component) => {
+const withRouting = (component) => {
```

Comment:
I know I suggested this name, but I think I'd like to use `withRouter` instead. We used that name for our HOC in v2, so users should already be familiar with it.